src/Avalonia.FuncUI/DSL/Base/IDataTemplateHost.fs

```diff
@@ -0,0 +1,49 @@
+namespace Avalonia.FuncUI.DSL
+
+[<AutoOpen>]
+module IDataTemplateHost =
+    open Avalonia.FuncUI.Types
+    open Avalonia.FuncUI.Builder
+    open Avalonia.Controls.Templates
+
+    module private Internals =
+        let patchDataTemplates(dataTemplates: DataTemplates) (templates: IDataTemplate seq) =
+
+            if Seq.isEmpty templates then
+                dataTemplates.Clear()            
+            else
+                templates
+                |> Seq.except dataTemplates
+                |> dataTemplates.RemoveAll
+
+                for newIndex, template in Seq.indexed templates do
+                    let oldIndex = dataTemplates |> Seq.tryFindIndex template.Equals
+
+                    match oldIndex with
+                    | Some oldIndex when oldIndex = newIndex -> ()
+                    | Some oldIndex -> dataTemplates.Move(oldIndex, newIndex)
+                    | None -> dataTemplates.Insert(newIndex, template)
+
```

Comment:
I implemented binding for Interfaces here, but would it be better to implement binding either in a normal class or in an abstract class?
It seems that Avalonia's plan is to reduce the number of Interfaces.
If so, I would like to implement binding for the Control class.

Reply:
> If so, I would like to implement binding for the Control class.

As in, make the binding functions against the ```Control``` class instead of the ```IDataTemplateHost``` interface?

Reply:
I would bind the actual interfaces 👍